From 714aefd0b5a73525eca82fe3dc85be6930ea0941 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Mon, 30 Dec 2013 15:21:05 +0800
Subject: [PATCH 8/9] ENGR00293455 fix The Total and Available space show 0B
 after erase sdcard.

the disk change event from driver indicates 0 dis number.
when part add event comes, it regards the whole sdcard as
one large partition. Actually, it has one boot partition
and one vfat partition.
reset the disk number to part number when it happens.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 DirectVolume.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/DirectVolume.cpp b/DirectVolume.cpp
index d5badd7..62b6d79 100644
--- a/DirectVolume.cpp
+++ b/DirectVolume.cpp
@@ -205,6 +205,10 @@ void DirectVolume::handlePartitionAdded(const char *devpath, NetlinkEvent *evt)
         part_num = 1;
     }
 
+    if (part_num == 1 && mDiskNumParts == 0) {
+        mDiskNumParts = part_num;
+    }
+
     if (part_num > MAX_PARTITIONS || part_num < 1) {
         SLOGE("Invalid 'PARTN' value");
         return;
-- 
1.8.0

